### PR TITLE
[Commerzbank] Werkpapierverkauf kann nicht importiert werden

### LINKED ISSUE #1470: [Commerzbank] Werkpapierverkauf kann nicht importiert werden
## Document
<details><summary>Wertpapierdokument_577844400_20210204 (1).pdf</summary><p>

```json
[
  [
    "Ihren",
    "Auftrag",
    "haben",
    "wir",
    "gemäß",
    "unseren",
    "Sonderbedingungen",
    "für",
    "Wertpapiergeschäfte",
    "wie",
    "nachstehend",
    "ausgeführt.",
    "Die",
    "Wertpapiere",
    "haben",
    "wir",
    "der",
    "Abrechnung",
    "entsprechend",
    "gebucht.",
    "Wir",
    "bitten",
    "Sie,",
    "diese",
    "Abrechnung",
    "auf",
    "ihre",
    "Richtigkeit",
    "und",
    "Vollständigkeit",
    "zu",
    "XXXXXX",
    "überprüfen",
    "und",
    "etwaige",
    "Einwendungen",
    "unverzüglich",
    "zu",
    "erheben.",
    "40300",
    "DUESSELDORF",
    "Telefon",
    ":",
    "02421",
    "183-0",
    "Filnr:",
    "000",
    "00000",
    "000",
    "Depotnr.:",
    "0000000",
    "00",
    "BLZ:",
    "000",
    "000",
    "00",
    "Xxxxx",
    "Xxxxxxxx",
    "Xxxxxxx",
    "Xxxxxxxx",
    "00",
    "00000",
    "Xxxxxxx",
    "GESCHÄFTSABRECHNUNG",
    "VOM",
    "04.02.2021",
    "für",
    "Ihr",
    "StartDepot",
    "*",
    "Wertpapierverkauf",
    "Geschäftsnummer",
    ":",
    "00",
    "000000",
    "000000000000-000",
    "Rechnungsnummer",
    ":",
    "000000000000X000",
    "Geschäftstag",
    ":",
    "04.02.2021",
    "Abwicklung",
    ":",
    "Rückgabe",
    "Handelszeit",
    ":",
    "19:59",
    "Uhr",
    "(MEZ/MESZ)",
    "Wertpapier-Bezeichnung",
    "Wertpapierkennnummer",
    "DWS",
    "Inv.-",
    "ESG",
    "Equity",
    "Income",
    "DWS2NY",
    "Inhaber-Anteile",
    "LD",
    "o.N.",
    "Nennwert",
    "Zum",
    "Preis",
    "von",
    "St.",
    "0,40",
    "EUR",
    "119,51",
    "Kurswert",
    ":",
    "EUR",
    "47,80",
    "IBAN",
    "Valuta",
    "Zu",
    "Ihren",
    "Gunsten",
    "vor",
    "Steuern",
    "XX00",
    "0000",
    "0000",
    "0000",
    "0000",
    "00",
    "EUR",
    "09.02.2021",
    "EUR",
    "47,80",
    "Verwahrungs-Art:",
    "GIROSAMMELDEPOT",
    "Ihre",
    "Wertpapier-Order",
    "wurde",
    "über",
    "das",
    "Online",
    "Banking",
    "erteilt.",
    "Diesem",
    "Wertpapiergeschäft",
    "ging",
    "keine",
    "individuelle",
    "Anlageempfehlung",
    "der",
    "Bank",
    "voraus.",
    "Informationen",
    "zur",
    "steuerlichen",
    "Behandlung",
    "dieses",
    "Geschäftsvorgangs",
    "und",
    "den",
    "auf",
    "Ihrem",
    "Konto",
    "gebuchten",
    "Endbetrag",
    "finden",
    "Sie",
    "auf",
    "der",
    "separaten",
    "Steuermitteilung.",
    "C",
    "O",
    "M",
    "M",
    "E",
    "R",
    "Z",
    "B",
    "A",
    "N",
    "K",
    "Aktiengesellschaft",
    "Diese",
    "Abrechnung",
    "wird",
    "von",
    "der",
    "Bank",
    "nicht",
    "unterschrieben",
    "Die",
    "Leistung",
    "ist",
    "gemäß",
    "§4",
    "Nr.8",
    "UStG",
    "umsatzsteuerfrei.",
    "USt-Id-Nr.:",
    "DE",
    "000",
    "000",
    "000",
    "X000",
    "XX000X/00/00/0000"
  ]
]

```

</p></details>

## Community profile

https://community.tresor.one/u/45380eed

---
Issue submitted with https://import.tresor.one

### PR DESCRIPTION
**Pull request by oxi0**
Hi,

I have built in the possibility of a sale for commerzbank documents.

That fixes #1470

### CODE REVIEW

File: src/brokers/commerzbank.js [hunk 1]
```diff
@@ -137,7 +139,16 @@ const parseSingleTransaction = textArr => {
...
+  } else if (isSell(textArr)) { 
+    type = 'Sell';
+    date = findDateBuy(textArr);
+    wkn = findWknBuy(textArr);
+    company = findCompanyBuy(textArr);
+    shares = +findSharesBuy(textArr);
+    amount = findAmountBuy(textArr);
+    price = findPriceBuy(textArr);
+    fee = findFeeBuy(textArr, amount);
+  }	else if (isDividend(textArr)) {
```
Comment (SirGibihm) [inline src/brokers/commerzbank.js:151]: You could change the function names to "findDateBuySell" but this is not a big issue at all

File: tests/brokers/__mocks__/commerzbank/sell/2021-DWS_Inv.json [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    "000000000000X000",
+    "Geschäftstag",
+    ":",
+    "04.02.2021",
+    "Abwicklung",
+    ":",
+    "Rückgabe",
+    "Handelszeit",
+    ":",
+    "19:59",
```
Comment (SirGibihm) [inline tests/brokers/__mocks__/commerzbank/sell/2021-DWS_Inv.json:90]: It seems to me like you could also parse the exact time of the transaction or am I misinterpreting this timestamp? `19:59` seems to be quite odd for a sell but might be correct.

**Review by SirGibihm — CHANGES_REQUESTED** [review]
Looking really good so far! I added one minor and one major remark. Just re-request reviews if you have made any changes (or disagree with my review :D )

File: src/brokers/commerzbank.js (hunk 1 above)
Comment (apentermann) [reply src/brokers/commerzbank.js:159]: Done

File: tests/brokers/__mocks__/commerzbank/sell/2021-DWS_Inv.json (hunk 2 above)
Comment (apentermann) [reply tests/brokers/__mocks__/commerzbank/sell/2021-DWS_Inv.json:90]: Done

File: src/brokers/commerzbank.js [hunk 3]
```diff
@@ -41,6 +41,12 @@ const findSharesDividend = textArr => {
 const findDateBuy = textArr =>
   textArr[textArr.findIndex(t => t.includes('Geschäftstag')) + 2];
 
+const findDateSell = textArr =>
+  textArr[textArr.findIndex(t => t.includes('Geschäftstag')) + 2];
+
+const findTimeSell = textArr =>
+  textArr[textArr.findIndex(t => t.includes('Handelszeit')) + 2];
+
```
Comment (SirGibihm) [inline src/brokers/commerzbank.js:49]: Sorry to be nitpicky here but you could join the functions `findDateBuy` and `findDateSell` as they do exactly the same thing: 
```javascript
const findDateBuySell = textArr =>
  textArr[textArr.findIndex(t => t.includes('Geschäftstag')) + 2];
```

**Review by SirGibihm — CHANGES_REQUESTED** [review]
One last thing

File: src/brokers/commerzbank.js (hunk 1 above)
Comment (oxi0) [reply src/brokers/commerzbank.js:156]: no problem, I thought it would be clearer that way.
I fixed it.

File: src/brokers/commerzbank.js (hunk 3 above)
Comment (oxi0) [reply src/brokers/commerzbank.js:46]: No problem, I thought it would be clearer that way.
I fixed it.

**Review by SirGibihm — APPROVED** [review]

**Comment by SirGibihm** [comment]
off we go! Thank you!

**Pull request merged**